### DISCUSSION TITLE
[develop] Discussion on how/where to provide platform-specific install instructions for CMAQ

### DISCUSSION THREAD

**Discussion by padhrigmccarthy** [ufs-community/ufs-srweather-app#712] [Q&A] [answered]
## Description
When installing CMAQ on cheyenne, there are a few platform-specific instructions, such as hashes to use in Externals.cfg (to pick up the newly-added cheyenne configs) and changes to devbuild.sh to disable UPP build compilation. Users may also find it helpful to have an example config.yaml that works on the platform. What would be the best way to provide these materials to users of CMAQ on specific platforms?

## Solution
I'm willing to provide cheyenne-specific 'getting started' instructions for CMAQ, and an example AQM community config for cheyenne, but would like to know the best way to provide these. For example, should I add a 'Cheyenne' section to the [AQM Wiki](https://github.com/ufs-community/ufs-srweather-app/wiki/Air-Quality-Modeling)? Should I check in a cheyenne-specific aqm.community config.yaml? I'm open to all suggestions.

## Requirements**
The goal of this work is to provide instructions and examples to CMAQ users on cheyenne that may not apply to other platforms. This could reduce the amount of trial-and-error for first time users.

**Comment by gspetro-NOAA** [comment]
Hello! 

I have converted your issue to a GitHub Discussion based on our lead code manager's suggestion. It seems like a discussion would be valuable to determine a best path forward, and then you or we can go ahead and add an issue if the path we settle on requires a pull request. I will reach out to members of my team and the CM team for comment, although there may be a bit of a delay because some people are on vacation (many schools are on April break this week or next). 

Thanks,
Gillian Petro, EPIC Advanced User Support (AUS) Team

**Comment by gspetro-NOAA** [comment]
Hello! 

I talked to my team earlier today, and I thought you might find the Code and Configuration Standards section of the [Contributor's Guide](https://github.com/ufs-community/ufs-srweather-app/wiki/Contributor's-Guide) useful as we work on a solution. I would be hesitant to post instructions to the wiki that use something other than the standard SRW/AQM hashes because that runs contrary to the Code and Configuration Standards. However, perhaps @MichaelLueken would be okay with temporary instructions for Cheyenne in the AQM wiki, as long as it is well-marked that this is an interim solution? I have asked him to mention this discussion at our Friday CM meeting to hear others' thoughts. Ultimately, we want users to be able to run on as many systems as possible, so the work you've been doing is a huge help in working toward that goal. @EdwardSnyder-NOAA had some ideas for adding AQM/Cheyenne configuration info to the SRW repo, so perhaps he can chime in with some ideas for how that could be done? That may be a better long-term solution. And ideally, we could expand that work to include additional RDHPCS (e.g., Jet, Gaea).

Best,
Gillian

**Comment by MichaelLueken** [comment, marked as answer]
Hi @padhrigmccarthy!

Currently, we are striving to maintain all hashes in a single Externals.cfg file.  For the the various component hashes that need to be updated to allow CMAQ to build and run on Cheyenne, are these hashes in the develop branch, or within forks?  If they are in the develop branches, then I would recommend working with @chan-hoo to update the hashes to the versions that include the modifications to build on Cheyenne (if they haven't been updated to that point yet) within the SRW App repo.  Once the Externals.cfg file has been updated, then I think that temporarily updating the current AQM wiki with instructions to build and run on Cheyenne would be fine.

Thanks,

Mike

**Comment by padhrigmccarthy** [comment]
Thank you, Mike!

**Comment by gspetro-NOAA** [comment]
Since Cheyenne was decommissioned on Dec. 31, 2023, this discussion is being closed as outdated. Feel free to post another discussion to continue the conversation for other platform and/or AQM support.
